File: 1_Two_Sum/brute_force.py
class Solution(object):
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        for i in range(n-1):
            a = nums[i]
            for j in range(i+1, n):
                b = nums[j]
                if a + b == target:
                    return i, j

# Time Complexity: O(n^2)
# Runtime: 4861 ms	Memory: 15 MB

File: 1_Two_Sum/brute_force.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int a, b;
        vector<int> indices;
        for (int i = 0; i < nums.size()-1; i++){
            a = nums[i];
            for (int j = i+1; j < nums.size(); j++){
                b = nums[j];
                if (a + b == target){
                    indices.push_back(i);
                    indices.push_back(j);
                    return indices;
                }
            }
        }
        return indices;
    }
};

// Time Complexity: O(n^2)
// Runtime: 410 ms	Memory: 10.2 MB